### PR TITLE
Ignore suspicious domain names and more careful work with links

### PR DESCRIPTION
**Pull request by koriaf**

### CODE REVIEW

File: crawler-node/src/crawler/worker.py [hunk 1]
```diff
@@ -300,15 +300,21 @@ def postprocess_resp(domain_name, scheme, url, resp, data, error, head_execution
...
+                    print("Ignoring the link {}, too long".format(normalized_url))
             else:
                 # external link
                 external_links.add(parsed_url.geturl())
                 if parsed_url.netloc != domain_name and parsed_url.netloc not in external_domains:
                     external_domains.add(parsed_url.netloc)
                     if parsed_url.netloc.endswith('.gov.au'):
                         # interesting for us
-                        put_to_redis("SEEN", parsed_url.netloc)
+                        if ":" in parsed_url.netloc or "@" in parsed_url.netloc:
```
Comment (monkeypants) [inline crawler-node/src/crawler/worker.py]: what about `https://thingything.gov.au:8080/welcome` ?

Comment (koriaf) [reply crawler-node/src/crawler/worker.py]: Still suspicious, but worth updating to handle these characters only in domain part... or even properly parse it to username/password/host/port/etc, if we find this useful

Comment (monkeypants) [reply crawler-node/src/crawler/worker.py]: As I read https://docs.python.org/3/library/urllib.parse.html, netloc shouldn't contain the port any of that junk. So it's really weird that our netloc would ever contain `:` or `@`. Are we passing urlparse something strange?

Comment (koriaf) [reply crawler-node/src/crawler/worker.py]: This is not so simple. And sometimes we have links like `href="https://email@address.domain" and broken links like this, so such code ignores them.
```
>>> urlparse("https://mailto:wow.xxx/")
ParseResult(scheme='https', netloc='mailto:wow.xxx', path='/', params='', query='', fragment='')
>>> urlparse("https://mailtome@wow.xxx")
ParseResult(scheme='https', netloc='mailtome@wow.xxx', path='', params='', query='', fragment='')

```

**Comment by koriaf** [comment]
Closing that PR in favour of #38, which includes this one

**Pull request closed without merging**